### PR TITLE
Use == and != for string comparisons in C#

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR replaces .Equals by == or != for string comparisons in C#.

It also replaces "" comparisons by Length comparisons (0 or > 0).

### CODE REVIEW

File: csharp/test/Ice/operations/Twoways.cs [hunk 1]
```diff
@@ -56,57 +56,57 @@ internal static void twoways(global::Test.TestHelper helper, Test.MyClassPrx p)
                 Ice.Communicator communicator = helper.communicator();
                 string[] literals = p.opStringLiterals();
 
-                test(Test.s0.value.Equals("\\") &&
+                test(Test.s0.value == "\\" &&
                      Test.s0.value.Equals(Test.sw0.value) &&
```
Comment (pepone) [inline csharp/test/Ice/operations/Twoways.cs:60]: Shouldn't we use `==` in all the other comparisons in the && conditions?

File: csharp/test/Ice/properties/Client.cs [hunk 2]
```diff
@@ -20,10 +20,10 @@ public override int
...
             Ice.Properties properties = communicator().getProperties();
-            test(properties.getProperty("Ice.Trace.Network").Equals("1"));
-            test(properties.getProperty("Ice.Trace.Protocol").Equals("1"));
-            test(properties.getProperty("Config.Path").Equals("./config/中国_client.config"));
-            test(properties.getProperty("Ice.ProgramName").Equals("PropertiesClient"));
+            test(properties.getProperty("Ice.Trace.Network") == "1");
+            test(properties.getProperty("Ice.Trace.Protocol") == "1");
+            test(properties.getProperty("Config.Path") == "./config/中国_client.config");
+            test(properties.getProperty("Ice.ProgramName") == "PropertiesClient");
             test(appName().Equals(properties.getProperty("Ice.ProgramName")));
```
Comment (pepone) [inline csharp/test/Ice/properties/Client.cs:27]: `appName()` returns a string too, doesn't it?

```
test(properties.getProperty("Ice.ProgramName") == appName());
```

**Review by pepone — APPROVED** [review]
Looks good

File: csharp/test/Ice/properties/Client.cs (hunk 2 above)
Comment (bernardnormier) [reply csharp/test/Ice/properties/Client.cs:27]: Yes, it was an automatic update. There are many more Equals that can be replaced by ==

**Pull request merged**